lib/handover.js

```diff
@@ -128,8 +134,16 @@ class Plugin {
 
       // If the Ethereum stuff came up empty, return the
       // HNS root server response unmodified.
-      if (!data || data.length === 0)
+      if (!data || data.length === 0) {
+        // never send referrals that end with .eth or ._eth
+        // since recursive may cache these referrals causing a servfail
+        // for future lookups
+        if(hasEnsReferral) {
```

Comment:
tiny nit: space after `if` -- this project doesn't have specific lint rules but hsd does so mine as well get a feel for it ;-)